core/src/main/java/org/apache/calcite/rel/rules/CoreRules.java

```diff
@@ -399,7 +399,7 @@ private CoreRules() {}
       MinusToDistinctRule.Config.DEFAULT.toRule();
 
   /** Rule to translates a {@link Minus} to {@link Join} anti-join}. */
-  public static final MinusToAntiJoinRule MINUS_TO_ANTI_JOIN_RULE =
+  public static final MinusToAntiJoinRule MINUS_TO_ANTI_JOIN =
```

Comment:
I don't think you can rename public constants, this will break user code that depends on these names.
Even if the name is not great, it's not worth the trouble changing it.

Reply:
Sorry, I've restored it.

Reply:
This was introduced after `1.39.0`, so it's an unreleased feature at the moment, even assuming someone is using it from a snapshot, I personally consider breaking changes only w.r.t. released versions, so for me we are still in time to change this before `1.40.0`, but that's my personal opinion, maybe someone could confirm or deny my assumption?

Reply:
@asolimando I started out with the same idea as you, but Mihai might be the one using snapshots, so I'm not sure what to do.

Reply:
Don’t worry about snapshots. Let’s keep the same name as in 1.39

Reply:
@mihaibudiu, according to git blame, this whole file was not there at all in `1.39.0`, it's been added later during the development of `1.40.0`, that's why I am suggesting it's fine (under this exceptional condition) to modify a public constant without considering it a breaking change IMO. Do you share the same feeling?

Reply:
Sure

Reply:
Thanks for the confirmation @mihaibudiu, @xiedeyantu if you feel like it you can change it back again

Reply:
I want to change the name back to the same style as the other rules. Can you help me see if there are any other changes needed? Can I just revert and squash it?

Reply:
You mean specifically for this change or in general? For this change, just do it through the IDE's refactoring features so there should be no discrepancies due to the renaming.

At that point you can safely squash as Mihai has already approved the PR (and this change), if tests are green the contribution will be good to be merged.

Reply:
Thanks for @mihaibudiu @asolimando, I have change it back again. And keep this pr in one commit.